src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/TreatWarningsAsErrorsRuleProvider.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.Build.Framework.XamlTypes;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Rules;
+
+/// <summary>
+/// Defines a <see cref="Rule"/> to extract information to support the <see cref="VBDiagnosticSeverityEnumProvider"/>.
+/// </summary>
+[ExportRuleObjectProvider(nameof(TreatWarningsAsErrorsRuleProvider), PropertyPageContexts.Project)]
+[AppliesTo(ProjectCapability.VisualBasic)]
+[Order(0)]
+internal sealed class TreatWarningsAsErrorsRuleProvider : IRuleObjectProvider
```

Comment:
I know I previously said I didn't want to introduce yet another way for us to ship `Rule` objects in our repo, but the truth is that defining this `Rule` in code was way easier than creating a new .xaml file and making sure it got plumbed through the build and setup authoring properly. I can still pull it out to a .xaml file if there are strong objections to this.

Reply:
I'm currently working on those build and setup authoring processes. I can confirm they are not as easy as just writing code.